fiftyone/brain/internal/core/uniqueness.py

```diff
@@ -66,83 +63,73 @@ def compute_uniqueness(
-    embeddings_field = None
-    if isinstance(similarity_backend, str) and not similarity_index_obj:
-        logger.info(
-            f"Initializing similarity index with {similarity_backend} backend and {similarity_index} brain_key."
-        )
-        similarity_index_obj = fb.compute_similarity(
-            samples,
-            backend=similarity_backend,
-            brain_key=similarity_index,
-            embeddings=False,
+    if roi_field is not None:
+        fov.validate_collection_label_fields(
+            samples, roi_field, _ALLOWED_ROI_FIELD_TYPES
         )
 
-        if roi_field is not None:
-            fov.validate_collection_label_fields(
-                samples, roi_field, _ALLOWED_ROI_FIELD_TYPES
-            )
-
-        if etau.is_str(embeddings):
-            embeddings_field, embeddings_exist = fbu.parse_embeddings_field(
-                samples,
-                embeddings,
-                patches_field=roi_field,
-            )
-            embeddings = None
-        else:
-            embeddings_exist = None
-
-        if model is None and embeddings is None and not embeddings_exist:
-            model = fbm.load_model(_DEFAULT_MODEL)
-            if batch_size is None:
-                batch_size = _DEFAULT_BATCH_SIZE
-
-        if roi_field is not None:
-            # @todo experiment with mean(), max(), abs().max(), etc
-            agg_fcn = lambda e: np.mean(e, axis=0)
-        else:
-            agg_fcn = None
-
-        embeddings, sample_ids, _ = fbu.get_embeddings(
+    if etau.is_str(embeddings):
+        embeddings_field, embeddings_exist = fbu.parse_embeddings_field(
             samples,
-            model=model,
-            model_kwargs=model_kwargs,
+            embeddings,
             patches_field=roi_field,
-            embeddings_field=embeddings_field,
-            embeddings=embeddings,
-            force_square=force_square,
-            alpha=alpha,
-            handle_missing="image",
-            agg_fcn=agg_fcn,
-            batch_size=batch_size,
-            num_workers=num_workers,
-            skip_failures=skip_failures,
-            progress=progress,
         )
-        similarity_index_obj.add_to_index(embeddings, sample_ids)
+        embeddings = None
+    else:
+        embeddings_field = None
+        embeddings_exist = None
+
+    if etau.is_str(similarity_index):
+        similarity_index = samples.load_brain_results(similarity_index)
+
+    if (
+        model is None
+        and embeddings is None
+        and similarity_index is None
+        and not embeddings_exist
+    ):
+        model = fbm.load_model(_DEFAULT_MODEL)
+        if batch_size is None:
+            batch_size = _DEFAULT_BATCH_SIZE
+
+    if roi_field is not None:
+        # @todo experiment with mean(), max(), abs().max(), etc
+        agg_fcn = lambda e: np.mean(e, axis=0)
+    else:
+        agg_fcn = None
+
+    embeddings, sample_ids, _ = fbu.get_embeddings(
```

Comment:
Unfortunately, `get_embeddings()` needs to be called in all cases, even when a pre-existing `similarity_index` is provided, because we need to query the index to see if a given sample ID actually exists in the vector index. It may not if the index was generated for a subset of the dataset. A typical case for this would be if new samples have been added to the dataset.

Reply:
Good point. IIUC this implementation will raise a warning about the missing samples and continue with the uniqueness computation for the subset of samples. What if a significant number of sample ids are missing from the provided similarity index? Do you still want this behavior?

Reply:
Yes I think its better to compute uniqueness for everything we can rather than raise a fatal error.